### PR TITLE
boards: arm: add nucleo_g070rb board

### PR DESCRIPTION
**Pull request by marinjurjevic**
Added ST Nucleo G070RB board.

### CODE REVIEW

**Comment by FRASTM** [comment]
check compliance 
and please run the samples/hello_world  to show the console output
else LGTM

**Comment by marinjurjevic** [comment]
Rebased and resolved compliance issues.

**Comment by erwango** [comment]
You need to update `tests/drivers/adc/adc_api/src/test_adc.c` with board reference

File: boards/arm/nucleo_g070rb/Kconfig.board [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# STM32 Nucleo-64 development board with STM32G070RB MCU configuration
+
+# Copyright (c) 2019 Philippe Retornaz <philippe@shapescale.com>
+# Copyright (c) 2019 STMicroelectronics
```
Comment (erwango) [inline boards/arm/nucleo_g070rb/Kconfig.board:4]: Use your own copyright here.

File: boards/arm/nucleo_g070rb/Kconfig.defconfig [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# STM32 Nucleo-64 development board with STM32G070RB MCU
+
+# Copyright (c) 2019 Philippe Retornaz <philippe@shapescale.com>
+# Copyright (c) 2019 STMicroelectronics
```
Comment (erwango) [inline boards/arm/nucleo_g070rb/Kconfig.defconfig:4]: Use your own copyright here.

File: boards/arm/nucleo_g070rb/arduino_r3_connector.dtsi [hunk 3]
```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2019 Linaro Limited
```
Comment (erwango) [inline boards/arm/nucleo_g070rb/arduino_r3_connector.dtsi:2]: Ditto

File: boards/arm/nucleo_g070rb/nucleo_g070rb.dts [hunk 4]
```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (c) 2019 Philippe Retornaz <philippe@shapescale.com>
+ * Copyright (c) 2019 STMicroelectronics
```
Comment (erwango) [inline boards/arm/nucleo_g070rb/nucleo_g070rb.dts:3]: Ditto

File: boards/arm/nucleo_g070rb/doc/index.rst [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+*************************
+
+Applications for the ``nucleo_g070rb`` board configuration can be built and
+flashed in the usual way (see :ref:`build_an_application` and
+:ref:`application_run` for more details).
+
+Flashing
+========
+
+Nucleo G070RB board includes an ST-LINK/V3 embedded debug tool interface.
```
Comment (erwango) [inline boards/arm/nucleo_g070rb/doc/index.rst:153]: ST-LINK/V2-1

File: boards/arm/nucleo_g070rb/nucleo_g070rb.yaml [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+  - uart
+  - gpio
+  - i2c
+  - spi
+  - counter
+  - watchdog
+  - pwm
+  - adc
+  - dac
+  - dma
```
Comment (erwango) [inline boards/arm/nucleo_g070rb/nucleo_g070rb.yaml:25]: DAC and DMA are not present in dts

**Review by erwango — CHANGES_REQUESTED** [review]

File: boards/arm/nucleo_g070rb/nucleo_g070rb.yaml (hunk 6 above)
Comment (marinjurjevic) [reply boards/arm/nucleo_g070rb/nucleo_g070rb.yaml]: I removed DAC.

Regarding DMA, `nucleo_g071rb` which I based this PR on, also does not have dma in dts file.
It's actually defined in root file `stm32g051.dtsi`, which `nucleo_g071rb.dts` indirectly includes.

The same holds for `nucleo_g070rb` which indirectly includes `stm32g050.dtsi`.

Knowing this, is dma node required in yaml file?

Comment (erwango) [reply boards/arm/nucleo_g070rb/nucleo_g070rb.yaml]: Adding dma will trigger dma test. So will require binding in dma tests. Not sure it is required, as this is already covered by `nucleo_g071rb`. We don't need each and single board to be tested/compiled

File: boards/arm/nucleo_g070rb/nucleo_g070rb.yaml [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+  - arduino_serial
+  - uart
+  - gpio
+  - i2c
+  - spi
+  - counter
+  - watchdog
+  - pwm
+  - adc
+  - dma
```
Comment (erwango) [inline boards/arm/nucleo_g070rb/nucleo_g070rb.yaml:24]: Please remove dma

**Review by erwango — CHANGES_REQUESTED** [review]

File: boards/arm/nucleo_g070rb/nucleo_g070rb.yaml (hunk 7 above)
Comment (marinjurjevic) [reply boards/arm/nucleo_g070rb/nucleo_g070rb.yaml]: Done.

**Review by erwango — APPROVED** [review]

**Review by gautierg-st — APPROVED** [review]

**Pull request merged**